app/models/concerns/course/assessment/questions_concern.rb

```diff
@@ -17,7 +17,23 @@ def attempt(submission)
                          map { |answer| [answer.question, answer] }.to_h
 
     map do |question|
-      attempting_answers.fetch(question) { question.attempt(submission) }
+      attempting_answers[question] || question.latest_correct_answer(submission) ||
```

Comment:
Sorry I don't quite get "have another method that checks for graded answers, attempting answers".  Do you mean that we get all the un-answered(or not correctly answered question) and only attempt them ?